### DISCUSSION TITLE
How do I implement a custom comparison?

### DISCUSSION THREAD

**Discussion by giorgiopogliani** [owen-it/laravel-auditing#956] [Q&A] [answered]
I have a decimal column and in my custom implementation I was doing `bccomp($value1, $value2, 2) == 0` to check if the values has changed. Can I do something similar with this package?

**Comment by giorgiopogliani** [comment]
Basically I am getting audits like:
- old values: `{"price":"8.20"}`
- new values: `{"price":8.2}`

**Reply by willpower232** [reply]
Have you cast the price column in your model in any particular way? or do you have some attribute method changing the formatting?

**Reply by parallels999** [reply]
Something like this
https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting
```php
protected function casts(): array
{
    return [
        'price' => 'double',
    ];
}
```

**Reply by giorgiopogliani** [reply, marked as answer]
I rounded the value to the same precision of the column before updating the model and seems to be working correctly
